openapi-cli/src/main/java/io/ballerina/generators/BallerinaSchemaGenerator.java

```diff
@@ -258,13 +257,33 @@ private TypeDefinitionNode getTypeDefinitionNode(OpenAPI openAPI,
                     typeDefinitionNodeList.add(typeDefNode);
                 }
             }
+        } else if (schemaValue.get$ref() != null) {
+            Token typeRef = AbstractNodeFactory.createIdentifierToken(getValidName(
+                    extractReferenceType(schemaValue.get$ref()), true));
+            Token asterisk = AbstractNodeFactory.createIdentifierToken("*");
+            Token semicolon = AbstractNodeFactory.createIdentifierToken(";");
+            TypeReferenceNode recordField =
+                    NodeFactory.createTypeReferenceNode(asterisk, typeRef, semicolon);
+            recordFieldList.add(recordField);
+            NodeList<Node> fieldNodes = AbstractNodeFactory.createNodeList(recordFieldList);
+            RecordTypeDescriptorNode recordTypeDescriptorNode =
+                    NodeFactory.createRecordTypeDescriptorNode(createToken(SyntaxKind.RECORD_KEYWORD),
+                            createToken(OPEN_BRACE_TOKEN), fieldNodes, null,
+                            createToken(SyntaxKind.CLOSE_BRACE_TOKEN));
+            MarkdownDocumentationNode documentationNode =
+                    createMarkdownDocumentationNode(createNodeList(schemaDoc));
+            MetadataNode metadataNode = createMetadataNode(documentationNode, createEmptyNodeList());
+            typeDefNode = NodeFactory.createTypeDefinitionNode(metadataNode,
+                    null, typeKeyWord, typeName, recordTypeDescriptorNode, createToken(SEMICOLON_TOKEN));
+            typeDefinitionNodeList.add(typeDefNode);
+
         } else if (schemaValue.getType() == null) {
 
             MarkdownDocumentationNode documentationNode =
                     createMarkdownDocumentationNode(createNodeList(schemaDoc));
             MetadataNode metadataNode =
                     createMetadataNode(documentationNode, createEmptyNodeList());
-            if (schemaValue.getNullable()) {
+            if (schemaValue.getNullable() != null) {
```

Comment:
```suggestion
            if (schemaValue.getNullable() != null) {
                if (schemaValue.getNullable()) {
                    typeDefNode = createTypeDefinitionNode(metadataNode,
                            null, createIdentifierToken("public type"),
                            createIdentifierToken(getValidName((schema.getKey()), true)),
                            createSimpleNameReferenceNode(createIdentifierToken(
                                    "any?")),
                            createToken(SEMICOLON_TOKEN));
                } else {
                    typeDefNode = createTypeDefinitionNode(metadataNode,
                            null, createIdentifierToken("public type"),
                            createIdentifierToken(getValidName((schema.getKey()), true)),
                            createSimpleNameReferenceNode(createIdentifierToken(
                                    "any")),
                            createToken(SEMICOLON_TOKEN));
                }
            }
```

Reply:
Shall we change the condition in the above way because if getNullable is not null, it can have two values `true` and `false`.